pallets/liquid-staking/src/lib.rs

```diff
@@ -1627,13 +1658,15 @@ pub mod pallet {
 
         #[require_transactional]
         pub fn do_matching() -> DispatchResult {
+            let remove_results = FastUnstakeRequests::<T>::clear(u32::max_value(), None);
```

Comment:
may not be safe to clear all here, better to add some limit(e.g:200)  and re-benchmark [this call](https://github.com/parallel-finance/parallel/blob/0d2a2995b2ab2b9f1e75fd98c39d8d3e47410155/pallets/liquid-staking/src/benchmarking.rs#L345-L361) based on the worst case.
Another option is to remove this storage and just add a flag(allow fast unstake or not) in `Unlockings` like what acala implement now

Reply:
we can check substrate staking pallet on how do they clear

Reply:
@yrong one concern for another option is: what if one user both do unstake and fast_unstake in the same era, then we can not distinguish the amount from `Unlockings`.
Acala has two storage: `RedeemRequests ` and `Unbondings `, once the era end, they will move all from `RedeemRequests` to `Unbondings`.

Reply:
@GopherJ I think the underlying code of remove_prefix also has a `limit` argument:
https://paritytech.github.io/substrate/master/frame_support/pallet_prelude/struct.StorageDoubleMap.html#method.remove_prefix
"if limit is none then all values in the client backend are deleted."

Reply:
> 

@MrPai then instead of clearing all `FastUnstakeRequests` when era advanced how about we also iter the storage to `do_fast_match_unstake` to match these request first before `do_matching`